presto-main/src/main/java/com/facebook/presto/operator/SpatialIndexBuilderOperator.java

```diff
@@ -101,10 +139,25 @@ public SpatialIndexBuilderOperator createOperator(DriverContext driverContext)
                     outputChannels,
                     indexChannel,
                     radiusChannel,
+                    partitionChannel,
                     spatialRelationshipTest,
                     filterFunctionFactory,
                     expectedPositions,
-                    pagesIndexFactory);
+                    pagesIndexFactory,
+                    getPartitions(driverContext.getPipelineContext().getTaskContext()));
+        }
+
+        private Map<Integer, Envelope2D> getPartitions(TaskContext taskContext)
+        {
+            if (spatialPartitions.isEmpty()) {
+                return ImmutableMap.of();
+            }
+
+            int partitionNumber = taskContext.getTaskId().getId();
+            int totalPartitions = taskContext.getTotalPartitions().getAsInt();
+            return spatialPartitions.entrySet().stream()
```

Comment:
Could you please put a comment why do we want to still filter the partitions? As far is i understand we are doing this just to decrease the memory footprint. Please correct me if i'm wrong.

Reply:
@arhimondr Originally, I expected each node to process exactly one partition. It wasn't necessary to propagate partition IDs for each build and probe row into the join, but the join operator needed to know which partition it is processing. Later, I realized that I need to support processing multiple spatial partitions on a single node and modified the code to propagate partition IDs. I didn't notice that the join operator no longer needs to know which partitions it is processing. This information could be used to perform some sanity checks on the probe and build row partitions, but otherwise is unnecessary.

That's a nice catch. Thank you. I assume you'd like me to simplify the code and just pass full list of spatial partitions around. Please, confirm and I'll update accordingly.

Reply:
Yeah, i think that adds an unnecessary complexity to already complex code. Please remove this if you don't mind.